eng/common/scripts/Test-Spelling.ps1

```diff
@@ -0,0 +1,91 @@
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [string] $TargetRef,
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $CspellConfigPath = "./.vscode/cspell.json"
+)
+
+. $PSScriptRoot/logging.ps1
+
+if ((Get-Command git | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate git. Install git https://git-scm.com/downloads"
+    exit 1
+}
+
+if ((Get-Command npx | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate npx. Install NodeJS (includes npm and npx) https://nodejs.org/en/download/"
+    exit 1
+}
+
+if (!(Test-Path $CspellConfigPath)) {
+    LogError "Could not locate config file $CspellConfigPath"
+    exit 1
+}
+
+$exitCode = 0
+
+# Lists names of files that were in some way changed between the 
+# current ref and $TargetRef. Excludes files that were deleted to
+# prevent errors in Resolve-Path
+Write-Host "git diff --diff-filter=d --name-only $TargetRef"
+$changedFiles = git diff --diff-filter=d --name-only $TargetRef `
```

Comment:
Just to ensure you only get the files you want you should pass HEAD as the other comparison point. Without that it would also potentially pick-up other changes to files in the working directory, etc. 
```suggestion
$changedFiles = git diff --diff-filter=d --name-only $TargetRef HEAD `
```

Reply:
2 parameters: 
* TargetRef -> rename to BaseRef
* Something to hold HEAD so we can insert it in DevOps